homeassistant/components/backup/manager.py

```diff
@@ -244,12 +244,21 @@ async def _async_upload_backup(
     ) -> None:
         """Upload a backup to selected agents."""
         LOGGER.warning("Uploading backup %s to agents %s", backup.backup_id, agent_ids)
+
+        async def send_backup() -> AsyncGenerator[bytes]:
+            with await self.hass.async_add_executor_job(path.open, "rb") as f:
+                while chunk := f.read(2**20):
```

Comment:
Is this async safe?

Reply:
No, probably not. I'll change it so the file reads are handed off to executors.